### ISSUE TITLE
RS0046 code fix introduces compilation errors when names are aliased

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn-analyzers#3813]
#### Analyzer package

Unknown

#### Package Version

3.3.0-beta1.20262.5

#### Diagnostic ID

RS0046

#### Repro steps

```csharp
void Method(TextSpan? selectionOpt)
{
  if (selectionOpt.HasValue)
  {
    var selection = selectionOpt.Value;
    Console.WriteLine(selection);
  }
}
```

Apply the code fix for RS0046.

#### Expected behavior

The resulting code compiles without errors.

#### Actual behavior

The code does not compile due to a conflict with the name `selection`.

**Comment by mavasani** [comment]
Dupe of https://github.com/dotnet/roslyn-analyzers/issues/3707. @Evangelink actually tried to implement it but couldn't figure out the right API usage - LookupSymbols seems to not provide existing symbols in scope.

**Comment by sharwell** [comment]
💭 I'm not sure this is a duplicate. It seems related but not exactly the same.

**Comment by Evangelink** [comment]
@sharwell @mavasani is there any one you would know who could help with the "correct" use of the method so we can quickly fix these cases?

**Comment by Evangelink** [comment]
I did some debugging to try and understand why I don't get the right results and I have noticed that even when called from a method parameter location the `LookupSymbols` method only returns namespace, class and members level symbols. I do manage to retrieve the other parameters and the local variable by passing the span of the method body.

Now the question is, shall I pass the method body/expression body as a diagnostic argument to avoid the "complex" logic in the code-fix or shall I do the detection from the code-fix.

Note that there is at least one more rule that will have to reuse the logic so depending on how we implement it we might be better off with a helper method.

**Comment by sharwell** [comment]
The first thing I would look at is how Roslyn's name suggestion completion provider avoids conflicts when suggesting names.

**Comment by Evangelink** [comment]
@sharwell Do you have some usage example? I don't know this api/feature.

**Comment by Evangelink** [comment]
Ping @sharwell
